src/pkg/scan/rest/v1/spec.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	// HTTPAcceptHeader represents the HTTP accept header
+	HTTPAcceptHeader = "Accept"
+	// HTTPContentType represents the HTTP content-type header
+	HTTPContentType = "Content-Type"
+	// MimeTypeOCIArtifact defines the mime type for OCI artifact
+	MimeTypeOCIArtifact = "application/vnd.oci.image.manifest.v1+json"
+	// MimeTypeDockerArtifact defines the mime type for docker artifact
+	MimeTypeDockerArtifact = "application/vnd.docker.distribution.manifest.v2+json"
+	// MimeTypeNativeReport defines the mime type for native report
+	MimeTypeNativeReport = "application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"
+	// MimeTypeRawReport defines the mime type for raw report
+	MimeTypeRawReport = "application/vnd.scanner.adapter.vuln.report.raw"
+	// MimeTypeAdapterMeta defines the mime type for adapter metadata
+	MimeTypeAdapterMeta = "application/vnd.scanner.adapter.metadata+json; version=1.0"
+	// MimeTypeScanRequest defines the mime type for scan request
+	MimeTypeScanRequest = "application/vnd.scanner.adapter.scan.request+json; version=1.0"
+	// MimeTypeScanResponse defines the mime type for scan response
+	MimeTypeScanResponse = "application/vnd.scanner.adapter.scan.response+json; version=1.0"
+)
+
+// RequestResolver is a function template to modify the API request, e.g: add headers
+type RequestResolver func(req *http.Request)
+
+// Definition for API
+type Definition struct {
```

Comment:
Do we really need `Definition` here?  In the client, some methods call `Resolver` but some methods do not, this is very strange.

Reply:
So far, all the methods have their resolvers. Even in the future, there are some methods not need resolvers, a nil resolver can also be ok. It is explicitly aligned with the API spec and easy for using.